lib/resque/job.rb

```diff
@@ -227,6 +256,20 @@ def args
       @payload['args']
     end
 
+    # Returns the metadata from the job's payload
+    def metadata
+      payload["meta"] ||= {}
+    end
+
+    # Returns the Time that this Job was queued, or nil.
+    def queued_at
+      @queued_at ||= metadata.key?("queued_at") ? Time.at(metadata["queued_at"]) : nil
+    end
+
+    def queued_at=(v)
+      metadata['queued_at'] = v.to_f
+    end
```

Comment:
FEATURE: `queued_at`

This uses the `metadata` feature to track when a job is enqueued.

Ported over from the monolith as it seemed generally applicable.

This could be PRed to master, but opting-in via a feature flag would likely be preferred.